File: src/main/java/com/github/cannor147/request/colorization/StraightColorizationScheme.java
package com.github.cannor147.request.colorization;

import com.github.cannor147.painter.RGBColor;

public class StraightColorizationScheme extends ColorizationScheme {
    @Override
    public RGBColor calculateColor(ColorizationParameter colorizationParameter) {
        return colorizationParameter.getColor().orElse(defaultColor).getRgbColor();
    }
}

File: src/main/java/com/github/cannor147/request/colorization/StraightColorizationScheme.kt
package com.github.cannor147.request.colorization

import com.github.cannor147.painter.RGBColor

class StraightColorizationScheme : ColorizationScheme() {
    override fun calculateColor(colorizationParameter: ColorizationParameter): RGBColor {
        return colorizationParameter.getColor().orElse(defaultColor).rgbColor
    }
}